scripts/servo_test.py

```diff
@@ -1,53 +1,151 @@
-#!/usr/bin/env python
-import pyudev
-from servo import Servo
+#!/usr/bin/env python3
+
+"""
+Servo board test script.
+
+This will:
+
+ - open up a servo board over USB
+ - initialise the switch mode power supply output
+ - sweep all servo outputs back and forth
+"""
+
 import time
+from typing import Any, Iterable, NamedTuple, Optional
+
+import usb.core
+import usb.util
+
+
+class USBDevice(NamedTuple):
+    """A single device found on USB."""
+
+    serial: str
+    node: usb.core.Device
+    busnum: int
+    devnum: int
+
+
+def _enumerate_usb_devices(*, product: int, vendor: int) -> Iterable[USBDevice]:
+    """Create a sorted list of USB devices of the given type."""
+    devices = list(usb.core.find(find_all=True, idProduct=product, idVendor=vendor))
+    devices.sort(key=lambda x: x.serial_number)
+    for device in devices:
```

Comment:
If we change this to return a list, then we don't need to do the list cast below. Having this as a generator yields no benefits (no pun intended)

Reply:
I agree that it might as well be a list, since I can't see a use case for this other than how it's used right now. If another use case appears that means it would make sense to be a generator it can be changed back. Even then, you'd have to have a LOT of USB devices attached for it to make any noticeable difference.

Either way, I'm going to merge this now and this change can be made in a further pull request if desired.